File: test_result.py
class TestResult:
    def __init__(self, function, unit_test):
        self.function = function
        self.unit_test = unit_test
        self.result = function.call_method(unit_test.arguments)
        self.passes = self.result == unit_test.expected

    def __str__(self):
        not_as_expected = '' if self.passes else f' != {self.unit_test.expected}'
        return f'{self.unit_test.arguments} -> {self.result}{not_as_expected}'

File: test_result.js
"use strict";
class TestResult {
    constructor(candidate, unitTest) {
        this.unitTest = unitTest;
        this.unitTest = unitTest;
        this.result = candidate.execute(unitTest.argumentList);
        this.passes = this.result === unitTest.expected;
    }
    toString() {
        return `${this.unitTest.argumentList} -> ${this.result}`;
    }
    toHtml() {
        return new Paragraph(this.toString());
    }
}
